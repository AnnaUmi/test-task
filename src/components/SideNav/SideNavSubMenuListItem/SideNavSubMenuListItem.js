import React from 'react';
import PropTypes from 'prop-types';
import Link from 'next/link';

import { SideNavSubMenuListemStyled } from '../SideNav.styles';

const SideNavSubMenuListem = ({ label, url, isActive }) => (
  <Link href={url}>
    <SideNavSubMenuListemStyled isActive={isActive}>{label}</SideNavSubMenuListemStyled>
  </Link>
);

SideNavSubMenuListem.propTypes = {
  label: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
  isActive: PropTypes.bool.isRequired,
};

export default SideNavSubMenuListem;
