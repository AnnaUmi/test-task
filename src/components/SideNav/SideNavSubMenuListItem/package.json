{
  "name": "side-nav-sub-menu-list-item",
  "version": "0.0.0",
  "private": true,
  "main": "./SideNavSubMenuListItem.js"
}
