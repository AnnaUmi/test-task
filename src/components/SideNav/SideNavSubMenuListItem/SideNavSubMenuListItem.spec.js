import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import SideNavSubMenuListItem from './SideNavSubMenuListItem';

describe('SideNavSubMenuListItem', () => {
  it('should match snapshot when not active', () => {
    const tree = renderer.create(
      <SideNavSubMenuListItem label="test label" url="test/url" isActive={false} />
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('should match snapshot when is active', () => {
    const tree = renderer.create(
      <SideNavSubMenuListItem label="test label" url="test/url" isActive />
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });
});
