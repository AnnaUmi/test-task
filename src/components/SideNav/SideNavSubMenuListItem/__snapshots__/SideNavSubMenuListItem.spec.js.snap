// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideNavSubMenuListItem should match snapshot when is active 1`] = `
.c0 {
  padding: 20px 60px;
  font-size: 18px;
  cursor: pointer;
  background-color: #7e7e7e;
  font-weight: bold;
}

.c0:hover {
  background-color: #bdbdbd;
}

<div
  className="c0"
  onClick={[Function]}
>
  test label
</div>
`;

exports[`SideNavSubMenuListItem should match snapshot when not active 1`] = `
.c0 {
  padding: 20px 60px;
  font-size: 18px;
  cursor: pointer;
}

.c0:hover {
  background-color: #bdbdbd;
}

<div
  className="c0"
  onClick={[Function]}
>
  test label
</div>
`;
